syntax = "proto3";

import "google/protobuf/empty.proto";

service ObjGen{
    rpc StartLoadObj(SatelliteAddr) returns (LoadResponse);
    rpc GetObjPos(Obj) returns (ObjPos);

    rpc SendObjPos(ObjPos) returns (google.protobuf.Empty);
}

message SatelliteAddr {
    int32 sat_id = 1;
    string sate_name = 2;
    string addr = 3;
    int32 port = 4;
}

message LoadResponse {
  bool successful = 1;
  int32 obj_id = 2;
}

message ObjPos {
    int32 obj_id = 1;
    string timestamp = 2;
    float delta_time = 3;
    float delta_lng = 4;
    float delta_lat = 5;
    float sog = 6;
    float cog = 7;
    float lng = 8;
    float lat = 9;
}

message Obj {
    int32 obj_id = 1;
}